cli/assets.go

```diff
@@ -256,3 +270,318 @@ func deleteAssetByIDCommand() *cobra.Command {
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			assetID := args[0]
+			ctx := client.SetMetadata(cmd.Context())
+			_, err = clnt.UnstarAsset(ctx, &compassv1beta1.UnstarAssetRequest{
+				AssetId: assetID,
+			})
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			fmt.Println(term.Bluef("Asset %v unstarred successfully", assetID))
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func starredAssetCommand() *cobra.Command {
+	var size, page uint32
+	var json string
+	cmd := &cobra.Command{
+		Use:   "starred",
+		Short: "list all the starred assets for current user",
+		Example: heredoc.Doc(`
+			$ compass asset starred
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			ctx := client.SetMetadata(cmd.Context())
+			res, err := clnt.GetMyStarredAssets(ctx, &compassv1beta1.GetMyStarredAssetsRequest{
+				Size:   size,
+				Offset: page,
+			})
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			if json != "json" {
+				report := [][]string{}
+				report = append(report, []string{"ID", "TYPE", "SERVICE", "URN", "NAME", "VERSION"})
+				index := 1
+				for _, i := range res.GetData() {
+					report = append(report, []string{i.Id, i.Type, i.Service, i.Urn, term.Bluef(i.Name), i.Version})
+					index++
+				}
+				printer.Table(os.Stdout, report)
+
+				fmt.Println(term.Cyanf("To view all the data in JSON format, use flag `-o json`"))
+			} else {
+				fmt.Println(term.Bluef(prettyPrint(res.GetData())))
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&json, "out", "o", "table", "flag to control output viewing, for json `-o json`")
+	cmd.Flags().Uint32Var(&size, "size", 20, "Size of each page")
+	cmd.Flags().Uint32Var(&page, "page", 0, "Number of pages")
+	return cmd
+}
+
+func versionHistoryAssetCommand() *cobra.Command {
+	var size, page uint32
+	cmd := &cobra.Command{
+		Use:   "versionhistory <id>",
+		Short: "get asset version history by id",
+		Example: heredoc.Doc(`
+			$ compass asset versionhistory <id>
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			assetID := args[0]
+			ctx := client.SetMetadata(cmd.Context())
+			res, err := clnt.GetAssetVersionHistory(ctx, &compassv1beta1.GetAssetVersionHistoryRequest{
+				Id:     assetID,
+				Size:   size,
+				Offset: page,
+			})
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			fmt.Println(term.Bluef(prettyPrint(res.GetData())))
+
+			return nil
+		},
+	}
+
+	cmd.Flags().Uint32Var(&size, "size", 20, "Size of each page")
+	cmd.Flags().Uint32Var(&page, "page", 0, "Number of pages")
```

Comment:
It's not really the number of pages, its the page number offset of results. So if we have total 60 results, default 0 will give us the first page of 3(60/20). Page 1 will give rows from 20-40.